src/apps/backend/modules/config/internals/config.py

```diff
@@ -0,0 +1,156 @@
+import os
+from pathlib import Path
+from typing import Any, Optional
+import yaml
+
+
+class Config:
+    _config_dict: dict[str, Any]
+    _config_path: Path 
+
+    @staticmethod
+    def load_config() -> None:
+        Config._intialize_config()
+        Config._process_custom_environment_variables()
+        print(yaml.dump(Config._config_dict))
+        
+    @staticmethod
+    def get(key: str, default: Optional[Any] = None) -> Any:
+        keys = key.split(".")
+        value = Config._config_dict
+        try:
+            for k in keys:
+                value = value[k]
+            return value
+        except (KeyError, TypeError):
```

Comment:
Can you add comments in the code when system would throw these errors (under what circumstances)?